HospitalSystem.ConsoleUI/Secretary/CheckupUI.cs

```diff
@@ -114,4 +118,138 @@ public void CheckRequests()
             ReadSanitizedLine();
         }
     }
-}
\ No newline at end of file
+
+    public void CreateCheckup(){
+
+        Console.Clear();
+        List<Patient> patients = _hospital.PatientService.GetPatients().ToList();
+
+        var firstName = EnterPatientFirstName();
+        var lastName = EnterPatientLastName();
+
+        CheckIfPatientExist(patients, firstName, lastName);
+        Patient patient = _hospital.PatientService.GetPatientByFullName(firstName, lastName);
+        
+        var referralPatient = patient.MedicalRecord.Referrals[0].Patient;
+        var referralDoctor = patient.MedicalRecord.Referrals[0].Doctor;
+        var referralAnamnesis =  patient.MedicalRecord.AnamnesisHistory;
+        var anamnesis = CreateSentence(referralAnamnesis);
+        var date = EnterDate();
+        var busyAppointments = _hospital.AppointmentService.GetCheckupsByDay(date);
+        busyAppointments.RemoveAll(c => c.Doctor.Id.ToString() != referralDoctor.Id.ToString());
+        var allAppointments = GetAllApointments(date);
+
+        Console.Clear();
+
+        RemoveBusyAppointments(busyAppointments, allAppointments);
+        RemovePastAppointments(allAppointments);
+        ShowFreeAppointments(allAppointments);
+
+        var appointment = EnterAppointment(allAppointments);
+        var dateTime = new DateTime(appointment.Year, appointment.Month, appointment.Day, appointment.Hour, appointment.Minute, appointment.Second);
+
+        Checkup check = new Checkup(dateTime, referralPatient, referralDoctor, anamnesis);
+        _hospital.AppointmentService.UpsertCheckup(check);
+
+        System.Console.Write("Successfully created appointment. Press anything to continue.");
+        ReadSanitizedLine();
```

Comment:
Extract methods. For example: Getting valid slots could be its own method

Reply:
It is not a big deal

Reply:
This function is like 60 lines